src/base/bittorrent/torrentinfo.cpp

```diff
@@ -213,24 +220,79 @@ QByteArray TorrentInfo::metadata() const
 
 QStringList TorrentInfo::filesForPiece(int pieceIndex) const
```

Comment:
I renamed the reverse function (piece range from file index). Probably you were confused by that. And yes, we need both functions.